### PR TITLE
Add new target Samsung s5js100

### PR DESCRIPTION
**Pull request by vince-zeng**
Add new target Samsung s5js100

### CODE REVIEW

File: pyocd/target/builtin/target_s5js100.py [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+                try:
+                    if not self.is_running():
+                        break
+                except exceptions.TransferError:
+                    self.flush()
+                    sleep(0.01)
+            else:
+                raise Exception("Timeout waiting for target halt")
+
+    def resume(self):
```
[the remark this answers is not in the document]
Comment (zxytddd) [reply pyocd/target/builtin/target_s5js100.py:278]: Yes, you are right. We will remove this overridden resume().
We used this overridden resume() for debug, but forgot to remove it.

File: src/gdb_test_program/linker_script.ld [hunk 2]
```diff
@@ -19,7 +19,7 @@
 MEMORY
 {
   /* Ram configurations are for smallest KL25 in family - 4K */
-  m_all        (rwx) : ORIGIN = 0x00000000, LENGTH = 0x600
+  m_all        (rwx) : ORIGIN = 0x00100000, LENGTH = 0x600
```
[the remark this answers is not in the document]
Comment (zxytddd) [reply src/gdb_test_program/linker_script.ld:22]: The commit 2adc409815202afd1946deea19bc5b5fcf96eeb4 shouldn't be included in the PR. I will remove this commit from the PR.

File: test/debug_context_test.py [hunk 3]
```diff
@@ -101,18 +101,18 @@ def debug_context_test(board_id):
         # Reproduce a gdbserver failure.
         print("\n------ Test 1: Mem cache ------")
         print("Writing gdb test binary")
-        ctx.write_memory_block8(0x20000000, gdb_test_binary_data)
+        ctx.write_memory_block8(0x406f4000, gdb_test_binary_data)
```
[the remark this answers is not in the document]
Comment (zxytddd) [reply test/debug_context_test.py:104]: The commit 2adc409815202afd1946deea19bc5b5fcf96eeb4 shouldn't be included in the PR. I will remove this commit from the PR.

[the remark this answers is not in the document]
Comment (vince-zeng) [reply test/debug_context_test.py]: I have reverted change https://github.com/mbedmicro/pyOCD/pull/869/commits/2adc409815202afd1946deea19bc5b5fcf96eeb4

File: pyocd/target/builtin/target_s5js100.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (vince-zeng) [reply pyocd/target/builtin/target_s5js100.py]: commit https://github.com/mbedmicro/pyOCD/pull/869/commits/dfecca2ef462527a49fdb56f2020e7d34877beb1

**Comment by vince-zeng** [comment]
- We reverted commit https://github.com/mbedmicro/pyOCD/pull/869/commits/2adc409815202afd1946deea19bc5b5fcf96eeb4  which changed linker_script.ld and debug_context_test.py by commit https://github.com/mbedmicro/pyOCD/pull/869/commits/105e3a39345d086b5afada87843af0b78ef9f659 
- Delete the override resume() in commit https://github.com/mbedmicro/pyOCD/pull/869/commits/dfecca2ef462527a49fdb56f2020e7d34877beb1
- Also add  new commit https://github.com/mbedmicro/pyOCD/pull/869/commits/f57aca1aa9456da43ab79fdd6fb0e0db6633c952

**Comment by vince-zeng** [comment]
Hi @flit , I can see the travis-ci build details now. It reports the following failure with python 2.7:
```
Command "python setup.py egg_info" failed with error code 1 in /home/travis/build/mbedmicro/pyOCD/
```
I can't reproduce this issue on my platform. Can I ignore the error?

**Comment by 0xc0170** [comment]
I started the test in the meantime

File: pyocd/target/builtin/target_s5js100.py [hunk 4]
```diff
@@ -0,0 +1,314 @@
+# pyOCD debugger
+# Copyright (c) 2006-2013 Arm Limited
```
Comment (flit) [inline pyocd/target/builtin/target_s5js100.py:2]: Please correct the copyright date to 2020.

File: pyocd/target/builtin/target_s5js100.py [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+
+class Flash_s5js100(Flash):
+    def __init__(self, target, flash_algo):
+        super(Flash_s5js100, self).__init__(target, flash_algo)
+        self._did_prepare_target = False
+        # LOG.info("S5JS100.Flash_s5js100.__init__ c")
+
+    def init(self, operation, address=None, clock=0, reset=True):
+        # LOG.info("S5JS100.Flash_s5js100.init c")
+        global is_flashing
```
Comment (flit) [inline pyocd/target/builtin/target_s5js100.py:128]: What is the `is_flashing` global used for? It doesn't seem to read anywhere.

File: pyocd/target/builtin/target_s5js100.py [hunk 6]
```diff
@@ -0,0 +1,314 @@
...
+
+
+class S5JS100(CoreSightTarget):
+    VENDOR = "Samsung"
+    AP_NUM = 0
+    ROM_ADDR = 0xE00FE000
+
+    memoryMap = MemoryMap(
+        # FlashRegion(    start=0x406f4000,  length=0x00100000, page_size = 0x400,     blocksize=4096, is_boot_memory=True, algo=flash_algo),
+        # FlashRegion(    start=0x406f4000,  length=0x00100000, page_size = 0x400, blocksize=0x1000, is_boot_memory=True, algo=flash_algo, flash_class=Flash_s5js100),
```
Comment (flit) [inline pyocd/target/builtin/target_s5js100.py:162]: Can you please remove these commented out lines?

File: pyocd/target/builtin/target_s5js100.py [hunk 7]
```diff
@@ -0,0 +1,314 @@
...
+                    if (dhcsr_reg & CortexM.S_RESET_ST) != 0:
+                        break
+                    sleep(0.1)
+                except exceptions.TransferError:
+                    self.flush()
+                    self._ap.dp.init()
+                    self._ap.dp.power_up_debug()
+                    sleep(0.01)
+            else:
+                raise Exception("Timeout waiting for reset")
```
Comment (flit) [inline pyocd/target/builtin/target_s5js100.py:250]: Please change to `pyocd.core.exceptions.TimeoutError`.

File: pyocd/target/builtin/target_s5js100.py [hunk 8]
```diff
@@ -0,0 +1,314 @@
...
+        with Timeout(5.0) as t_o:
+            while t_o.check():
+                try:
+                    if not self.is_running():
+                        break
+                except exceptions.TransferError:
+                    self.flush()
+                    sleep(0.01)
+            else:
+                raise Exception("Timeout waiting for target halt")
```
Comment (flit) [inline pyocd/target/builtin/target_s5js100.py:276]: Please change to `pyocd.core.exceptions.TimeoutError`.

**Comment by vince-zeng** [comment]
Hi @flit, we have completed the code clean up, and squash them to a single commit.  Please help to review again. Thanks a lot!

**Pull request merged**